pkg/deploy/generator/resources_rp.go

```diff
@@ -441,7 +441,19 @@ func (g *generator) rpVMSS() *arm.Resource {
 		"''')\n'",
 	)
 
-	trailer := base64.StdEncoding.EncodeToString(scriptRpVMSS)
+	var sb strings.Builder
+
+	// VMSS extensions only support one custom script
+	// Because of this, the util-*.sh scripts are prefixed to the bootstrapping script
+	// main is called at the end of the bootstrapping script, so appending them will not work
+	sb.WriteString(string(scriptUtilCommon))
+	sb.WriteString(string(scriptUtilPackages))
+	sb.WriteString(string(scriptUtilServices))
+	sb.WriteString(string(scriptUtilSystem))
+	sb.WriteString("\n#Start of rpVMSS.sh\n")
+	sb.WriteString(string(scriptRpVMSS))
+
+	trailer := base64.StdEncoding.EncodeToString([]byte(sb.String()))
```

Comment:
Alternatively, we could change the `go:embed` definitions for these scripts in scripts.go to be `string` directly instead of `[]byte` - but I agree with appending to a `[]byte` directly if possible, to keep the number of conversions to a minimum.

Reply:
I used string  builder because they are in `[]byte` format via `go:embed`. I hadn't really thought about embedding them directly as strings.  
I'll give that a try.

Reply:
Embeding them as strings then concatenating them works the same. I pushed a commit with this change.